opencost/opencost
# Fix error for AWS IRSA + Spot Data Feed Users

## What does this PR change?
* Set empty string for `awsServiceKeyName`.
* Previously, having placeholder text in that JSON forced OpenCost to disregard IRSA and look for a nonexistent ServiceKey.
* Also set empty string for prefix, in case users don't specify a prefix.

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* Fixes a bug for IRSA users

## Does this PR address any GitHub or Zendesk issues?
* Fixes https://github.com/opencost/opencost/issues/1955
* Fixes https://github.com/kubecost/cost-analyzer-helm-chart/issues/2238

## How was this PR tested?

1. Set up an EKS cluster
2. Install Kubecost with [Spot Data Feed configured](https://docs.kubecost.com/install-and-configure/install/cloud-integration/aws-cloud-integrations/aws-spot-instances#spot-data-feed-integration).
4. Set up IRSA
5. Restart the pod
6. See the following error

    ```
    WRN Skipping AWS spot data download: operation error S3: ListObjects, https response error StatusCode: 403, RequestID: AA4V2X6YQ2FJVN2F, HostID: +zg6hpFYP8vpA/+Ix0lq9EGkNz/s3B699oVC0MVeLZ7eY5bTH3nsleGd9KGpDiL3nMFPxAus+mE=, api error InvalidAccessKeyId: The AWS Access Key Id you provided does not exist in our records.
    ```

To fix the issue:

1. Set `"awsServiceKeyName: ""` in `/var/configs/aws.json`
2. Restart the pod
3. Success:

    ```
    2023-06-02T23:38:32.700867222Z INF Found spot info for: i-06552693f1ab7bd2e
    2023-06-02T23:38:32.70091437Z INF Found spot info for: i-09d5e95bdac0403d2
    2023-06-02T23:38:32.774288673Z INF Found spot info for: i-06552693f1ab7bd2e
    2023-06-02T23:38:32.774334968Z INF Found spot info for: i-09d5e95bdac0403d2
    2023-06-02T23:38:32.845723989Z INF Found spot info for: i-06552693f1ab7bd2e
    ```

## Does this PR require changes to documentation?
* No

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
* v1.104
